Linked_List/StartOfLoop.java

```diff
@@ -0,0 +1,111 @@
+/*
+* Find first node of a loop in a Linked List
+* A linked list is reported to form a loop. We have to find the first element of the loop or cycle.
+*
+*/
+
+import java.util.Scanner;
+
+class StartOfLoop
+{
+    static Node head;
+    static class Node
+    {
+        int data;
+        Node next;
+        Node(int d)
+        {
+            data = d;
+            next = null;
+        }
+    }
+
+    static void insert(int d)
+    {
+        Node n = new Node(d);
+        if(head == null)
+            head = n;
+        else
+        {
+            Node current = head;
+            while(current.next != null)
+                current = current.next;
+            current.next = n;
+        }
+    }
+    
+    static void createLoop(int p)
+    {
+        // For demonstration, we will link the last node with the element at position p.
+        Node current = head, tail = head;
+        while(tail.next != null)
+            tail = tail.next;
+        
+        int count = 1;
+        while(current.next != null && count < p)
+		{   
```

Comment:
Please fix indentation for while loop. The { should start from below the while statement